### ISSUE TITLE
PSNR is only 5.61

### ISSUE THREAD

**Issue by Beilinlin** [jinlab-imvr/Deform3DGS#3]
Dear Author,
I followed your training steps and ran 3000 iterations on the EndoNeRF dataset, but the resulting PSNR is only 5.61. Could you please teach me why this might be the case? I would greatly appreciate your assistance.
Thank you.
![1720879295710](https://github.com/user-attachments/assets/895a12d2-f0ad-4062-846c-2754347cb20f)

**Comment by lala-sean** [comment]
Hi, thanks for your attention to this repo. I tried to recreate the environment and re-run the code for pulling sequence. But the outcome looks fine to me (as figure below).
![image](https://github.com/user-attachments/assets/9749d122-83f8-4ae0-819f-a960bc6f0a97)
Could you elaborate more on your hardware device and software environment e.g., driver, cuda version so that we can try to reproduce the issue? If you are looking for a quick debugging, I may suggest to:
1. Re-run the code 
2. Re-run the code with a changed seed
3. Test the code on a different sequence like cutting
4. Make sure 'pip install -e submodules/depth-diff-gaussian-rasterization' is performed for rasterizer installation.
5. Remove the initialization by commenting [Line 257](https://github.com/jinlab-imvr/Deform3DGS/blob/02be2db4ad3517fa10e723486d0b151f93f53000/scene/endo_loader.py#L257) in Deform3DGS/scene/endo_loader.py

Feel free to let us know if any issues. Thanks!

**Comment by Beilinlin** [comment]
Dear author,
I apologize for bothering you again.
I have tried re-running the code, changing the seed, testing the code on a different sequence like cutting, and removing the initialization by commenting Line 257 in Deform3DGS/scene/endo_loader.py, but the results are still not satisfactory. The PSNR is still around 5, and the loss remains as high as 0.49.
I have ensured that 'pip install -e submodules/depth-diff-gaussian-rasterization' was performed for rasterizer installation.
![ENWB9@N6Dear author,
I apologize for bothering you again.
I have tried re-running the code, changing the seed, testing the code on a different sequence like cutting, and removing the initialization by commenting Line 257 in Deform3DGS/scene/endo_loader.py, but the results are still not satisfactory. The PSNR is still around 5, and the loss remains as high as 0.49.
![}0$9(K_TLL6FE1~ }86UJ}5](https://github.com/user-attachments/assets/508c488a-cd7a-4f85-8ba9-8f26bf05e516)

I have ensured that 'pip install -e submodules/depth-diff-gaussian-rasterization' was performed for rasterizer installation.
Y{HP%MJ0I {R2](https://github.com/user-attachments/assets/59ee86e6-6a62-4ab2-9b55-4b3660d630bd)

My setup includes an NVIDIA GeForce RTX 4090 GPU with 24564MiB of memory. My CUDA version is 11.7, and my Python version is 3.7.16.
Thank you for your guidance and support.

**Comment by Beilinlin** [comment]
Dear author,
I apologize for bothering you again.

I have tried re-running the code, changing the seed, testing the code on a different sequence like cutting, and removing the initialization by commenting Line 257 in Deform3DGS/scene/endo_loader.py, but the results are still not satisfactory. The PSNR is still around 5, and the loss remains as high as 0.49.
![}0$9(K_TLL6FE1~ }86UJ}5](https://github.com/user-attachments/assets/82ddbafb-4932-4bda-ac53-06ba7001756b)

I have ensured that 'pip install -e submodules/depth-diff-gaussian-rasterization' was performed for rasterizer installation.
Y{HP%MJ0I {R2](https://github.com/user-attachments/assets/b32ea8d8-6722-4047-801e-4a5593ddd2cd)

My setup includes an NVIDIA GeForce RTX 4090 GPU with 24564MiB of memory. My CUDA version is 11.7, and my Python version is 3.7.16.
Thank you for your guidance and support.

**Comment by lala-sean** [comment]
> Dear author, I apologize for bothering you again.
> 
> I have tried re-running the code, changing the seed, testing the code on a different sequence like cutting, and removing the initialization by commenting Line 257 in Deform3DGS/scene/endo_loader.py, but the results are still not satisfactory. The PSNR is still around 5, and the loss remains as high as 0.49. ![}0$9(K_TLL6FE1~ }86UJ}5](https://private-user-images.githubusercontent.com/147405303/348710010-82ddbafb-4932-4bda-ac53-06ba7001756b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjEwNDgxMjcsIm5iZiI6MTcyMTA0NzgyNywicGF0aCI6Ii8xNDc0MDUzMDMvMzQ4NzEwMDEwLTgyZGRiYWZiLTQ5MzItNGJkYS1hYzUzLTA2YmE3MDAxNzU2Yi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNzE1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDcxNVQxMjUwMjdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0wYjJmMWEzZDQ1ODQxOTE4N2JkNDVhM2MzNjJhNzY1ZTYwMDBjM2NiNjczOGRiOWE0ZTBmOTkwMzcxZjk4OWRhJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9._vVD4hWjolwsFD7H8VIeSbQpqTLwB2kZXnDt_HNuZq0)
> 
> I have ensured that 'pip install -e submodules/depth-diff-gaussian-rasterization' was performed for rasterizer installation. Y{HP%MJ0I {R2](https://github.com/user-attachments/assets/b32ea8d8-6722-4047-801e-4a5593ddd2cd)
> 
> My setup includes an NVIDIA GeForce RTX 4090 GPU with 24564MiB of memory. My CUDA version is 11.7, and my Python version is 3.7.16. Thank you for your guidance and support.

Thanks for your update! The issue encountered is very new to us. The code worked fine on A5000/A6000 with cuda-11.7 also. Can you have one more try to remove the previous submodules/depth-diff-gaussian-rasterization and install the most up-to-date rasterizer with depth supervision (https://github.com/ingra14m/depth-diff-gaussian-rasterization/tree/depth)? If it still cannot solve the issue, would you mind if we have an online meeting and troubleshooting together? Seems a little bit inefficient to use the GitHub issue page to troubleshoot. Welcome to drop me an email at s.yang@u.nus.edu. Thank you!

**Comment by Beilinlin** [comment]
Hi,

Thank you for your suggestions. After troubleshooting, I identified the issue. Due to limited server permissions, running the git submodule update --init --recursive command resulted in errors, which impacted the PSNR. To resolve this, I manually downloaded the problematic submodules. As a result, the situation has significantly improved.

Thank you again for your help!

**Comment by lala-sean** [comment]
> Hi,
> 
> Thank you for your suggestions. After troubleshooting, I identified the issue. Due to limited server permissions, running the git submodule update --init --recursive command resulted in errors, which impacted the PSNR. To resolve this, I manually downloaded the problematic submodules. As a result, the situation has significantly improved.
> 
> Thank you again for your help!

Good to know the reason. Thanks!
